loadtest/user/userentity/actions.go

```diff
@@ -110,7 +124,50 @@ func (ue *UserEntity) authOpenID(action authOpenIDAction) error {
 		return fmt.Errorf("%s failed with status code %d", action, loginResponse.StatusCode)
 	}
 
-	cookies := jar.Cookies(loginResponse.Request.URL)
+	// Perform an extra step when doing SAML authentication, since the Keycloak server won't redirect
+	// to the Mattermost server automatically, instead it will display a form with the SAML response
+	// that needs to be submitted to the Mattermost server. This is done via Javascript in the browser,
+	// but we can simulate it by parsing the form and doing the same request manually.
+	if provider == AuthenticationTypeSAML {
+		samlResponseBody, err := io.ReadAll(loginResponse.Body)
+		if err != nil {
+			return fmt.Errorf("error while reading saml response body: %w", err)
+		}
+		loginResponse.Body.Close()
+
+		redirectURLMatcher := keycloakIDPLoginFormActionRegex.FindSubmatch(samlResponseBody)
+		if len(redirectURLMatcher) == 0 {
+			return errors.New("redirect URL not found in SAML login page, there was probably an error or the configuration is wrong")
```

Comment:
Maybe we can be a bit more clear in the error mentioning that it's also possible that the UI has changed which has broken the regex?

Reply:
👍